jenkins/promotion/release-promotion.jenkinsfile

```diff
@@ -0,0 +1,433 @@
+            }
+        }
+        stage('OpenSearch Dashboards Apt promotion') {
+            agent {
+                docker {
+                    label AGENT_LINUX_X64
+                    image 'docker/library/alpine:3'
+                    registryUrl 'https://public.ecr.aws/'
+                    alwaysPull true
+                }
+            }
+            steps {
+                echo 'Triggering distribution-promote-repos for OpenSearch Dashboards apt'
+                build job: 'distribution-promote-repos', wait: true, parameters: [choice(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch-dashboards'), 
+                                                                                    choice(name: 'DISTRIBUTION_REPO_TYPE', value: 'apt'),
+                                                                                    string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_DASHBOARDS_RC),
+                                                                                    string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-dashboards-${params.RELEASE_VERSION}.yml"),
+                                                                                ]
+                echo 'Job distribution-promote-repos for OpenSearch Dashboards completed!'
+            }
+        }
+        stage('OpenSearch Dashboards Yum promotion') {
+            agent {
+                docker {
+                    label AGENT_LINUX_X64
+                    image 'docker/library/alpine:3'
+                    registryUrl 'https://public.ecr.aws/'
+                    alwaysPull true
+                }
+            }
+            steps {
+                echo 'Triggering distribution-promote-repos for OpenSearch Dashboards yum'
+                build job: 'distribution-promote-repos', wait: true, parameters: [choice(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch-dashboards'), 
+                                                                                    choice(name: 'DISTRIBUTION_REPO_TYPE', value: 'yum'),
+                                                                                    string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_DASHBOARDS_RC),
+                                                                                    string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-dashboards-${params.RELEASE_VERSION}.yml"),
+                                                                                ]
+                echo 'Job distribution-promote-repos for OpenSearch Dashboards yum completed!'
+            }
+        }
+        stage('OpenSearch Windows promotion') {
+            agent {
+                docker {
+                    label AGENT_LINUX_X64
+                    image 'docker/library/alpine:3'
+                    registryUrl 'https://public.ecr.aws/'
+                    alwaysPull true
+                }
```

Comment:
LGTM overall.  Since the docker agent is same for all the triggered runs I would recommend we loop through the lists [OS, OSD], [tar, deb, rpm, windows, apt, yum] w.r.t. [x64, arm64] for all the possible mappings instead of individually passing parameters for each run, as part of optimization in 2nd iteration.

Reply:
Those combinations are never unique @Divyaasm, also If I loop do you think I can still visualize as individual stage ? My idea was each stage should an individual promotion job, would be easy debug/verify if a promotion job is failed and run it manually as we do today.

Reply:
I see, currently we have implemented this parallelly in build and validation workflows. Need to figure out ways to do that if we opt to loop the possible mappings.